core/modules/field/field.block.inc

```diff
@@ -80,9 +80,28 @@ class FieldBlock extends Block {
     list($entity_type, $field_name) = explode('-', $this->childDelta);
     $entity = $this->contexts[$entity_type]->data;
 
-    // Do not render if the entity is not found on this layout.
-    if (empty($entity)) {
-      return NULL;
+    // If the entity is not an EntityInterface object, try to get one.
+    if (!is_a($entity, 'EntityInterface')) {
+      $info = entity_get_info($entity_type);
+      $id_key = $info['entity keys']['id'];
+      if (isset($entity->{$id_key})) {
+        $entity = $info['load hook']($entity->{$id_key});
+        if (is_a($entity, 'EntityInterface')) {
+          // Save the EntityInterface object for future use.
+          $this->contexts[$entity_type]->setData($entity);
+        }
+        else {
+          // Either there is a problem with this entity type definition,
+          //  or the entity does not exist or is failing to load for some reason.
+          //  Do not proceed.
+          return NULL;
+        }
+      }
+      else {
+        // Unable to load an entity without an id property value.
+        //  Do not proceed.
+        return NULL;
+      }
```

Comment:
```suggestion
      if (!isset($entity->{$id_key})) {
        // Unable to load an entity without an id property value.
        return NULL;
      }

      // Try load the entity from the hook.
      $entity = $info['load hook']($entity->{$id_key});

      if (!is_a($entity, 'EntityInterface')) {
        // Either there is a problem with this entity type definition,
        // or the entity does not exist or is failing to load for some reason.
        return NULL;
      }

      // Save the EntityInterface object for future use.
      $this->contexts[$entity_type]->setData($entity);
```

@bradbulger looking better. I still think it can be made clearer a bit more by using "guard clauses": make the conditionals flatter and avoiding "else" where possible. See https://codeinphp.github.io/post/replace-nested-conditions-with-guard-clauses/

Reply:
That's done.